### PR TITLE
Menu design revision

### PR DESCRIPTION
**Pull request by silviyaboteva**
Implements: https://github.com/telerik/kendo-themes-private/issues/266

### CODE REVIEW

File: packages/default/scss/menu/_layout.scss [hunk 1]
```diff
@@ -35,6 +34,10 @@
         flex: none;
         position: relative;
         user-select: none;
+
+        &.k-selected {
+            font-weight: var( --kendo-menu-item-selected-font-weight, #{$kendo-menu-item-selected-font-weight} );
```
Comment (inikolova) [inline packages/default/scss/menu/_layout.scss:39]: The font weight here should be set in the following way:
font-weight: $kendo-menu-item-selected-font-weight;

File: packages/html/src/menu/menu-scrollable-wrapper.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+export const MENU_SCROLLABLE_WRAPPER_CLASS = `k-menu-scroll-wrapper`;
+
+const states = [];
+
+const options = {};
+
+export type KendoMenuScrollableWrapperProps = {
+    orientation?: string;
+    wrapperStyles?: React.CSSProperties;
```
Comment (inikolova) [inline packages/html/src/menu/menu-scrollable-wrapper.spec.tsx:11]: Should we expose wrapperStyles prop? Could we use style attribute?

File: packages/bootstrap/scss/menu/_variables.scss
Comment (epetrow) [inline packages/bootstrap/scss/menu/_variables.scss:1]: Please make sure you are consistent in the way you are documenting the variables, for example:

`Width of the border around the menu` vs `The background of the menu`

You can check other documented components for reference - https://github.com/telerik/kendo-themes/blob/develop/packages/default/scss/button/_variables.scss

Note that this applies to all the themes.

File: packages/classic/scss/menu/_variables.scss [hunk 3]
```diff
@@ -2,51 +2,148 @@
 
 $kendo-menu-padding-x: k-spacing(0) !default;
```
Comment (epetrow) [inline packages/classic/scss/menu/_variables.scss:3]: Why are we not documenting these variables? Are they unused, if so maybe they can be removed?

File: packages/default/scss/menu/_variables.scss [hunk 4]
```diff
@@ -2,51 +2,149 @@
 
 $kendo-menu-padding-x: k-spacing(0) !default;
```
Comment (epetrow) [inline packages/default/scss/menu/_variables.scss:3]: Same question here, why are we not documenting these?

File: packages/bootstrap/scss/menu/_variables.scss [hunk 5]
```diff
@@ -1,49 +1,147 @@
...
+
+/// The inline margin of the horizontal menu separator.
+/// @group menu
+$kendo-menu-separator-margin-inline: k-spacing(1);
+/// The block margin of the horizontal menu separator.
+/// @group menu
+$kendo-menu-separator-margin-block: k-spacing(2);;
+/// The border color of  menu separator.
+/// @group menu
+$kendo-menu-separator-border: var(--kendo-color-border, #{$kendo-component-border}) !default;
```
Comment (epetrow) [inline packages/bootstrap/scss/menu/_variables.scss:142]: I think this should be `if($kendo-enable-color-system, k-color(border), $kendo-component-border)`

File: packages/classic/scss/menu/_variables.scss [hunk 6]
```diff
@@ -2,51 +2,148 @@
...
+
+/// The inline margin of the horizontal menu separator.
+/// @group menu
+$kendo-menu-separator-margin-inline: k-spacing(1);
+/// The block margin of the horizontal menu separator.
+/// @group menu
+$kendo-menu-separator-margin-block: k-spacing(1.5);
+/// The border color of  menu separator.
+/// @group menu
+$kendo-menu-separator-border: var(--kendo-color-border, #{$kendo-component-border}) !default;
```
Comment (epetrow) [inline packages/classic/scss/menu/_variables.scss:144]: Same here, maybe it should be `if($kendo-enable-color-system, k-color(border), $kendo-component-border)`

File: packages/default/scss/menu/_variables.scss [hunk 7]
```diff
@@ -2,51 +2,149 @@
...
+
+/// The inline margin of the horizontal menu separator.
+/// @group menu
+$kendo-menu-separator-margin-inline: k-spacing(1);
+/// The block margin of the horizontal menu separator.
+/// @group menu
+$kendo-menu-separator-margin-block: k-spacing(1.5);
+/// The border color of  menu separator.
+/// @group menu
+$kendo-menu-separator-border: var(--kendo-color-border, #{$kendo-component-border}) !default;
```
Comment (epetrow) [inline packages/default/scss/menu/_variables.scss:145]: Should it be `if($kendo-enable-color-system, k-color(border), $kendo-component-border)`

File: packages/fluent/scss/menu/_layout.scss [hunk 8]
```diff
@@ -100,10 +104,11 @@
         }
 
         > .k-separator {
-            margin: 0 var( --kendo-menu-separator-margin, #{$kendo-menu-separator-margin} );
+            margin-block: $kendo-menu-separator-margin-block;
```
Comment (epetrow) [inline packages/fluent/scss/menu/_layout.scss:107]: Add css variables here: 

`var(--kendo-menu-separator-margin-block, #{$kendo-menu-separator-margin-block})`
`var(--kendo-menu-separator-margin-inline, #{$kendo-menu-separator-margin-inline})`

File: packages/fluent/scss/menu/_layout.scss [hunk 9]
```diff
@@ -178,7 +188,8 @@
         }
 
         .k-separator {
-            margin: var( --kendo-menu-separator-margin, $kendo-menu-separator-margin ) 0;
+            margin-block: $kendo-menu-separator-margin-inline;
```
Comment (epetrow) [inline packages/fluent/scss/menu/_layout.scss:191]: Add css variable

File: packages/fluent/scss/menu/_theme.scss [hunk 10]
```diff
@@ -27,20 +27,31 @@
...
+            }
         }
 
         .k-link > .k-icon {
             color: var( --kendo-menu-icon-color, $kendo-menu-icon-color );
         }
 
+        > .k-separator {
+            @include fill(
+                $border: $kendo-menu-separator-border
```
Comment (epetrow) [inline packages/fluent/scss/menu/_theme.scss:51]: Add css variable

File: packages/fluent/scss/menu/_theme.scss [hunk 11]
```diff
@@ -88,6 +99,12 @@
                 }
             }
         }
+
+        > .k-separator {
+            @include fill(
+                $border: $kendo-menu-separator-border
```
Comment (epetrow) [inline packages/fluent/scss/menu/_theme.scss:105]: Same here add css variable

File: packages/html/src/menu/templates/menu-scrollable.tsx
Comment (epetrow) [inline packages/html/src/menu/templates/menu-scrollable.tsx:1]: We do not add component states like active, focus, hover, etc. in the templates. Please remove them from all templates.

Also I think you don't need to specifically pass`scrollButtonPosition` and ` wrapperStyles`, you could access them through  `{...props}`

File: packages/html/src/menu/tests/menu.tsx [hunk 12]
```diff
@@ -23,27 +23,26 @@ export default () =>(
...
+                                    <MenuListItem text="Focus" focus></MenuListItem>
+                                    <MenuListItem last text="Focus & Active" active focus></MenuListItem>
+                                </MenuList>
+                            </Popup>
+                        </>
+                    }></MenuItem>
+                    <MenuItem text="Selected" selected showArrow arrowIconName="caret-alt-down"></MenuItem>
                     <MenuItem text="Focus" focus></MenuItem>
-                    <MenuItem last text="Normal" icon="folder"></MenuItem>
+                    <MenuItem text="Disabled" disabled></MenuItem>
```
Comment (epetrow) [inline packages/html/src/menu/tests/menu.tsx:45]: Why are we  removing `last` here?

File: packages/html/src/menu/tests/menu.tsx [hunk 13]
```diff
@@ -73,10 +71,10 @@ export default () =>(
                                     </MenuList>
                                 </Popup>
                             </>
-                        }
-                    />
+                        }></MenuItem>
+                    <MenuItem text="Selected" selected showArrow arrowIconName="caret-alt-down"></MenuItem>
                     <MenuItem text="Focus" focus></MenuItem>
-                    <MenuItem last text="Normal" icon="folder"></MenuItem>
+                    <MenuItem text="Disabled" disabled></MenuItem>
```
Comment (epetrow) [inline packages/html/src/menu/tests/menu.tsx:77]: And here, `last` missing?

File: packages/material/scss/menu/_variables.scss [hunk 14]
```diff
@@ -1,51 +1,149 @@
 // Menu
 $kendo-menu-padding-x: k-spacing(0) !default;
```
Comment (epetrow) [inline packages/material/scss/menu/_variables.scss:2]: Is there a reason why these are not documented?

File: packages/material/scss/menu/_variables.scss [hunk 15]
```diff
@@ -1,51 +1,149 @@
...
+
+/// The inline margin of the horizontal menu separator.
+/// @group menu
+$kendo-menu-separator-margin-inline: k-spacing(1);
+/// The block margin of the horizontal menu separator.
+/// @group menu
+$kendo-menu-separator-margin-block: k-spacing(1);
+/// The border color of  menu separator.
+/// @group menu
+$kendo-menu-separator-border: var(--kendo-color-border, #{$kendo-component-border}) !default;
```
Comment (epetrow) [inline packages/material/scss/menu/_variables.scss:144]: I think this should be changed to `if($kendo-enable-color-system, k-color(border), $kendo-component-border)`

File: tests/_output/bootstrap/menu/menu-scrollable-buttons-alignment.png
Comment (epetrow) [inline tests/_output/bootstrap/menu/menu-scrollable-buttons-alignment.png:1]: Shouldn't the scrollable buttons in rtl start from right and end on left as these are the de facto start and end possitions?

File: tests/_output/bootstrap/spreadsheet/spreadsheet-tools.png
Comment (epetrow) [inline tests/_output/bootstrap/spreadsheet/spreadsheet-tools.png:1]: There seems to be something wrong with this test, could you check?

**Review by epetrow — DISMISSED** [review]

File: packages/classic/scss/menu/_variables.scss (hunk 3 above)
Comment (silviyaboteva) [reply packages/classic/scss/menu/_variables.scss:3]: Yes, they were unused. Will be removed.

File: packages/html/src/menu/templates/menu-scrollable.tsx
Comment (silviyaboteva) [reply packages/html/src/menu/templates/menu-scrollable.tsx:1]: If I don't specify it either passing just props to the component and then scrollButtonsPosition={props.buttonsPosition} wrapperStyles={props.wrapperStyles}, when I use the template in tests, the props don't seem to apply.

File: packages/html/src/menu/menu-scrollable-wrapper.spec.tsx [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+export const MENU_SCROLLABLE_WRAPPER_CLASS = `k-menu-scroll-wrapper`;
+
+const states = [];
+
+const options = {};
+
+export type KendoMenuScrollableWrapperProps = {
+    orientation?: string;
+    style?: React.CSSProperties;
```
Comment (inikolova) [inline packages/html/src/menu/menu-scrollable-wrapper.spec.tsx:11]: Do we need this _style_ prop? We usually set styles through _style attribute_ directly

File: packages/fluent/scss/menu/_variables.scss [hunk 17]
```diff
@@ -3,169 +3,196 @@
...
-/// The background color of focused menu items.
+/// The background color of focused Menu item.
 /// @group menu
 $kendo-menu-item-focus-bg: $kendo-menu-item-bg !default;
-/// The outline of focused menu items.
+
+/// The outline of focused Menu item.
 /// @group menu
 $kendo-menu-item-focus-outline: var( --kendo-focus-outline, none ) !default;
+$kendo-menu-item-focus-outline-offset: calc( #{k-spacing(0.5)} * -1 ) !default;
```
Comment (inikolova) [inline packages/fluent/scss/menu/_variables.scss:85]: This (and the following 2) variable is not documented

File: packages/default/scss/menu/_layout.scss [hunk 18]
```diff
@@ -107,6 +110,7 @@
     // Orientation -- vertical
     .k-menu-vertical {
         flex-direction: column;
+        width: 100%;
```
Comment (inikolova) [inline packages/default/scss/menu/_layout.scss:113]: In my opinion this _width_ property is not necessary

File: packages/default/scss/menu/_layout.scss [hunk 19]
```diff
@@ -150,6 +153,10 @@
         *::after {
             box-sizing: border-box;
         }
+
+        .k-menu-item {
+            font-weight: initial;
```
Comment (inikolova) [inline packages/default/scss/menu/_layout.scss:158]: We don't change the menu items font weight, so I think this rule is redundant

File: packages/default/scss/menu/_layout.scss (hunk 18 above)
Comment (silviyaboteva) [reply packages/default/scss/menu/_layout.scss:113]: it is necessary as the items shrink in vertical orientation :(

File: packages/fluent/scss/menu/_variables.scss (hunk 17 above)
Comment (silviyaboteva) [reply packages/fluent/scss/menu/_variables.scss:85]: I checked in other places where we document only the first (global) variable. For example in the Button: https://github.com/telerik/kendo-themes/blob/develop/packages/default/scss/button/_variables.scss#L224-L234

When we can document only the default variable and when to document them all?

File: packages/default/scss/menu/_layout.scss (hunk 18 above)
Comment (inikolova) [reply packages/default/scss/menu/_layout.scss:113]: Update -> it is needed for the scrollable vertical menu

File: packages/html/src/menu/templates/menu-scrollable.tsx [hunk 20]
```diff
@@ -1,18 +1,13 @@
...
-import { Button } from "../../button";
 
-export const MenuScrollable = ({ wrapperStyles, ...other }: any) => (
-    <div className="k-menu-scroll-wrapper" style={wrapperStyles}>
-        <Menu children={
+export const MenuScrollable = (props) => (
+        <Menu scrollable scrollButtonsPosition={props.buttonsPosition} wrapperStyles={props.wrapperStyles} children={
             <>
-                <MenuItem first text="Item 1"></MenuItem>
+                <MenuItem first text="Item 1" dir={props.dir}></MenuItem>
```
[the remark this answers is not in the document]
Comment (silviyaboteva) [reply packages/html/src/menu/templates/menu-scrollable.tsx:6]: it's leftover

File: packages/fluent/scss/menu/_layout.scss [hunk 21]
```diff
@@ -100,10 +104,11 @@
         }
 
         > .k-separator {
-            margin: 0 var( --kendo-menu-separator-margin, #{$kendo-menu-separator-margin} );
+            margin-block: var(--kendo-menu-separator-margin-block, #{$kendo-menu-separator-margin-block});
+            margin-inline: var(--kendo-menu-separator-margin-inline, #{$kendo-menu-separator-margin-inline});
             width: 0;
             height: auto;
-            border-bottom-width: var( --kendo-menu-separator-width, $kendo-menu-separator-width);
+            border-left-width: var( --kendo-menu-separator-width, $kendo-menu-separator-width);
```
Comment (inikolova) [inline packages/fluent/scss/menu/_layout.scss:111]: The syntax here should be: border-left-width: var( --kendo-menu-separator-width, #{$kendo-menu-separator-width});

File: packages/fluent/scss/menu/_layout.scss [hunk 22]
```diff
@@ -178,7 +189,8 @@
         }
 
         .k-separator {
-            margin: var( --kendo-menu-separator-margin, $kendo-menu-separator-margin ) 0;
+            margin-block: $kendo-menu-separator-margin-inline;
+            margin-inline: 0;
             height: 0;
             border-width: 0;
             border-top-width: var( --kendo-menu-separator-width, $kendo-menu-separator-width);
```
Comment (inikolova) [inline packages/fluent/scss/menu/_layout.scss:196]: The syntax here should be: border-top-width: var( --kendo-menu-separator-width, #{$kendo-menu-separator-width});

File: packages/fluent/scss/menu/_theme.scss (hunk 11 above)
Comment (inikolova) [reply packages/fluent/scss/menu/_theme.scss:105]: The syntax here should be: $border: var( --kendo-menu-separator-border, #{$kendo-menu-separator-border} )

File: packages/html/src/menu/tests/menu-scrollable-buttons-alignment-rtl.tsx [hunk 23]
```diff
@@ -0,0 +1,44 @@
+import { MenuScrollable, MenuVerticalScrollable } from '..';
+
+export default () =>(
+    <>
+        <div id="test-area" className="k-d-grid k-grid-cols-3">
+            <span>Horizontal Scrolling Menu</span>
```
Comment (inikolova) [inline packages/html/src/menu/tests/menu-scrollable-buttons-alignment-rtl.tsx:6]: Please add RTL here -> for example "Horizontal Scrolling Menu RTL"

**Comment by inikolova** [comment]
@silviyaboteva great work :) Some additional suggestions:

- This is not related to the current changes, however I believe it will be beneficial. In my opinion if there is "dir" prop it should suggests "rtl" or "ltr"; currently it is from type string -> https://github.com/telerik/kendo-themes/blob/menu-revision/packages/html/src/menu/menu-list-item.spec.tsx#L25
- In the "menu-vertical" test we could add selected state as well;

**Review by inikolova — APPROVED** [review]

**Pull request merged**